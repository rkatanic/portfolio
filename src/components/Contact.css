.contact {
  flex: 1;
}

.contact .icon {
  fill: var(--color-primary);
  height: 2.5rem;
  margin-right: var(--margin-1);
}

.contact-title {
  text-align: center;
  font-size: var(--font-size-3);
}

.contact-subtitle {
  text-align: center;
  color: var(--color-text);
}

.contact-container {
  display: flex;
  flex-wrap: wrap;
}

.contact-links {
  margin: var(--margin-1) 0;
  flex: 1;
}

.contact-link {
  display: flex;
  align-items: center;
}

.contact-link-info {
  margin: var(--margin-0-5);
  flex: 1;
}

.contact-link-info h3 {
  color: var(--color-title);
  margin: 0.125rem 0;
}

.contact-link a {
  text-decoration: underline;
  color: var(--color-text);
}

.contact-link a:hover {
  color: var(--color-primary);
}

@media screen and (max-width: 768px) {
  .contact-container {
    flex-direction: column;
  }

  .contact-form button {
    width: 100%;
    justify-content: center;
  }
}
