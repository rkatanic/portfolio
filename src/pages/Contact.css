#contact {
  display: flex;
  gap: 2rem;
  align-items: center;
}

.bubble {
  border-radius: 50%;
  text-align: center;
  flex: 1;
}
