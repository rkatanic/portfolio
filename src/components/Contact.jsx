import { useState } from "react";
import Section from "./Section";
import { ReactComponent as EmailIcon } from "../assets/icons/email.svg";
import { ReactComponent as GithubIcon } from "../assets/icons/github.svg";
import { ReactComponent as LinkedinIcon } from "../assets/icons/linkedin.svg";
import emailjs from "emailjs-com";
import { Formik, Form, Field, ErrorMessage } from "formik";
import * as Yup from "yup";

import "./Contact.css";

const Contact = () => {
  const validationSchema = Yup.object({
    name: Yup.string()
      .min(2, "Name must be atleast 2 characters long")
      .required("Name is required"),
    email: Yup.string()
      .email("Invalid email format")
      .required("Email is required"),
    subject: Yup.string()
      .min(4, "Subject must be atleast 4 characters long")
      .required("Subject is required"),
    message: Yup.string()
      .min(10, "Message must be atleast 10 characters long")
      .max(500, "Message can not be longer than 500 characters")
      .required("Message is required"),
  });

  const formikInitialValues = {
    name: "",
    email: "",
    subject: "",
    message: "",
  };

  const handleEmailSend = async (values, setSubmitting) => {
    console.log(values);
    try {
      setSubmitting(true);
      await emailjs.send(
        process.env.REACT_APP_EMAIL_SERVICE_ID,
        process.env.REACT_APP_EMAIL_TEMPLATE_ID,
        values,
        process.env.REACT_APP_EMAIL_USER_ID
      );
      setSubmitting(false);
    } catch (err) {
      alert(err.text);
      setSubmitting(false);
    }
  };

  return (
    <Section id="contact">
      <div className="contact">
        <h2 className="contact-title">Contact</h2>
        <h3 className="contact-subtitle">Don't hesitate to ask anything</h3>

        <div className="contact-container">
          <div className="contact-links">
            <div className="contact-link">
              <EmailIcon />
              <div className="contact-link-info">
                <h3>Email</h3>
                <p>k.rade313@gmail.com</p>
              </div>
            </div>
            <div className="contact-link">
              <LinkedinIcon />
              <div className="contact-link-info">
                <h3>LinkedIn</h3>
                <p>k.rade313@gmail.com</p>
              </div>
            </div>
            <div className="contact-link">
              <GithubIcon />
              <div className="contact-link-info">
                <h3>GitHub</h3>
                <p>k.rade313@gmail.com</p>
              </div>
            </div>
          </div>
          <Formik
            initialValues={formikInitialValues}
            validationSchema={validationSchema}
            onSubmit={(values, { setSubmitting }) =>
              handleEmailSend(values, setSubmitting)
            }
          >
            {({ isValid, dirty, values, errors, touched, isSubmitting }) => (
              <Form className="contact-form">
                <Field
                  placeholder="Enter your name"
                  type="name"
                  name="name"
                  id="name"
                  value={values.name}
                  className={errors.name && touched.name ? "input-error" : null}
                />
                <ErrorMessage name="name" />
                <Field
                  placeholder="Enter your email"
                  type="email"
                  email="email"
                  id="email"
                  name="email"
                  value={values.email}
                  className={
                    errors.email && touched.email ? "input-error" : null
                  }
                />
                <ErrorMessage name="email" />
                <Field
                  placeholder="Enter your subject"
                  type="subject"
                  subject="subject"
                  id="subject"
                  name="subject"
                  value={values.subject}
                  className={
                    errors.subject && touched.subject ? "input-error" : null
                  }
                />
                <ErrorMessage name="subject" />
                <Field
                  as="textarea"
                  placeholder="Enter your message"
                  type="message"
                  message="message"
                  id="message"
                  name="message"
                  value={values.message}
                  className={
                    errors.message && touched.message ? "input-error" : null
                  }
                />
                <ErrorMessage name="message" />
                <button
                  type="submit"
                  disabled={!isValid || isSubmitting}
                  className="btn btn-primary"
                >
                  {isSubmitting ? "Sending ..." : "Send email"}
                </button>
              </Form>
            )}
          </Formik>
        </div>
      </div>
    </Section>
  );
};

export default Contact;
