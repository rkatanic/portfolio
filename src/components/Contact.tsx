import Section from "./Section";
import ContactForm from "./contact/ContactForm";

import "./Contact.css";

const Contact = (): JSX.Element => (
  <Section id="contact">
    <h3 className="headline">Don't hesitate to ask anything</h3>
    <ContactForm />
  </Section>
);

export default Contact;
