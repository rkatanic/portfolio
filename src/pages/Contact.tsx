import Section from "../components/Section";
import ContactForm from "../components/contact/ContactForm";

import "./Contact.css";
import Card from "../components/Card";
import Scroll from "../components/Scroll";
import { Suspense, useRef } from "react";
import { Canvas, useFrame } from "@react-three/fiber";

const Contact = (): JSX.Element => {
  const Box = () => {
    const boxRef = useRef<any>();

    useFrame(() => {
      boxRef.current.rotation.y += 0.01;
    });

    return (
      <mesh
        ref={boxRef}
        rotation-x={Math.PI * 0.25}
        rotation-y={Math.PI * 0.25}
      >
        <icosahedronBufferGeometry args={[1.85]} />
        <meshBasicMaterial wireframe color={"hsl(0,0%,30%)"} />
      </mesh>
    );
  };
  return (
    <Scroll>
      <h1 className="section-title txt-lg txt-800">
        <span className="txt-outlined">0</span>3
      </h1>
      <h3 className="side-section-title txt-xxs txt-400">Get In Touch</h3>

      <div className="contact-content div-parent">
        <Canvas className="child-shape">
          <Suspense fallback={null}>
            <pointLight position={[5, 5, 5]} />
            <Box />
          </Suspense>
        </Canvas>
        <div className="child-txt">
          <div className="contact-heading">
            <h2 className="txt-xl txt-800">
              <span className="txt-outlined">HA</span>VE{" "}
              <span className="txt-outlined">A</span>
            </h2>
            <h2 className="txt-xl txt-800">
              <span className="txt-outlined">#</span>QUE
              <span className="txt-outlined">S</span>{" "}
            </h2>
            <h2 className="txt-xl txt-800">
              <div className="dash"></div>
              TIO
              <span className="txt-outlined">N?</span>
            </h2>
          </div>
        </div>
      </div>

      <ContactForm />
    </Scroll>
  );
};

export default Contact;
