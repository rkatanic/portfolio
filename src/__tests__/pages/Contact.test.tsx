import { render } from "@testing-library/react";
import Contact from "../../pages/Contact";

describe("Contact", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<Contact />);

    expect(baseElement).toMatchSnapshot();
  });
});
