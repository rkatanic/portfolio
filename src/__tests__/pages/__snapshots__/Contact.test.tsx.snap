// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should render 1`] = `
<body>
  <div>
    <section
      class="section"
      id="contact"
    >
      <h3
        class="headline"
      >
        Still have a question
        <span
          class="primary"
        >
          ?
        </span>
         Reach me out
      </h3>
      <form
        action="#"
        class="contact-form"
      >
        <div
          class="form-field"
        >
          <div
            class="form-field-label"
          >
            <label
              for="name"
            >
              name
              <span
                class="primary"
              >
                *
              </span>
            </label>
          </div>
          <input
            aria-label="name"
            class=""
            maxlength="50"
            name="name"
            placeholder="Enter your name"
            value=""
          />
          <div
            class="form-field-length-indicator"
          >
            <span
              class=""
            >
              0
            </span>
             
            / 
            <span>
              50
            </span>
          </div>
        </div>
        <div
          class="form-field"
        >
          <div
            class="form-field-label"
          >
            <label
              for="email"
            >
              email
              <span
                class="primary"
              >
                *
              </span>
            </label>
          </div>
          <input
            aria-label="email"
            class=""
            maxlength="100"
            name="email"
            placeholder="Enter your email"
            value=""
          />
          <div
            class="form-field-length-indicator"
          >
            <span
              class=""
            >
              0
            </span>
             
            / 
            <span>
              100
            </span>
          </div>
        </div>
        <div
          class="form-field"
        >
          <div
            class="form-field-label"
          >
            <label
              for="subject"
            >
              subject
              <span
                class="primary"
              >
                *
              </span>
            </label>
          </div>
          <input
            aria-label="subject"
            class=""
            maxlength="100"
            name="subject"
            placeholder="Enter your subject"
            value=""
          />
          <div
            class="form-field-length-indicator"
          >
            <span
              class=""
            >
              0
            </span>
             
            / 
            <span>
              100
            </span>
          </div>
        </div>
        <div
          class="form-field"
        >
          <div
            class="form-field-label"
          >
            <label
              for="message"
            >
              message
              <span
                class="primary"
              >
                *
              </span>
            </label>
          </div>
          <textarea
            aria-label="message"
            class=""
            maxlength="500"
            name="message"
            placeholder="Enter your message"
            rows="10"
          />
          <div
            class="form-field-length-indicator"
          >
            <span
              class=""
            >
              0
            </span>
             
            / 
            <span>
              500
            </span>
          </div>
        </div>
        <button
          class="btn btn-primary"
          type="submit"
        >
          Send email
        </button>
      </form>
    </section>
  </div>
  <div
    class="ReactModalPortal"
  />
</body>
`;
